zstacklib/zstacklib/utils/gpu.py

```
@@ -996,3 +1015,549 @@ def parse_alibaba_ppu_metric_output(output):
+        return result
+
+    return {"isDriverLoaded": False}
+
+
+def _collect_tianshu_legacy(pci_address):
+    """Tianshu legacy collection"""
+    r, o, e = bash_roe("which ixsmi")
+    if r != 0:
+        return {"isDriverLoaded": False}
+
+    if shell.run(is_tianshu_v1()) == 0:
```

Ruff (0.14.13)

`shell` may be undefined, or defined from star imports

(F405)